src/gobuploadservice/compare.py

```diff
@@ -35,39 +24,27 @@ def compare(msg):
     message = ImportMessage(msg)
     metadata = message.metadata
 
-    # Todo: this should be done elsewhere
-    base = init_storage(metadata, engine, Base)
-
     # Read new content into dictionary
     new_entities = {data['_source_id']: data for data in msg["contents"]}
 
-    # Get the table where the current entities are stored
-    DbEntity = getattr(base.classes, metadata.entity)
-
     # Get all current non-deleted ids for this source
-    session = Session(engine)
-    current_ids = session.query(DbEntity._source_id).filter_by(_source=metadata.source,
-                                                               _date_deleted=None).all()
-
-    # find deletes by comparing current ids to new entities
-    # if a current_id is not found in the new_entities it is interpreted as a deletion
-    deleted = {current._source_id: None for current in current_ids if current._source_id not in new_entities}
-
-    all = {**new_entities, **deleted}
-    # Make sure there are no id's (keys) that are present in both
-    assert len(all) == len(new_entities) + len(deleted)
-
-    events = []
-    for entity_id, data in all.items():
-        entity = session.query(DbEntity).filter_by(_source=metadata.source,
-                                                   _source_id=entity_id,
-                                                   _date_deleted=None).one_or_none()
-
-        modifications = get_modifications(entity, data, metadata.model)
-        event = get_event_for(entity, data, metadata, modifications)
-        events.append(event)
-
-    session.close()
+    db = GOBStorageHandler(metadata)
+    with db.get_session():
+        current_ids = db.get_current_ids()
+
+        # find deletes by comparing current ids to new entities
+        # if a current_id is not found in the new_entities it is interpreted as a deletion
+        deleted = {current._source_id: None for current in current_ids if current._source_id not in new_entities}
+
+        # combine new and deleted into one set
+        all = {**new_entities, **deleted}
+
+        events = []
```

Comment:
Probably add some comments about what is done in the next lines

Reply:
Done